src/github.com/getlantern/flashlight/ui/service.go

```diff
@@ -0,0 +1,102 @@
+package ui
+
+import (
+	"encoding/json"
+	"sync"
+)
+
+type envelopeType struct {
+	Type string
+}
+
+type Service struct {
+	Name    string
+	In      chan []byte
+	Out     chan []byte
+	helloFn func(func([]byte) error) error
+}
+
+var (
+	mu               sync.Mutex
+	defaultUIChannel *UIChannel
+
+	out      = make(chan []byte, 10)
+	services = make(map[string]*Service)
+)
+
+func (s *Service) watch() {
+	// Watch for new messages and sent them to the combined output.
+	for b := range s.Out {
+		out <- b
+	}
+}
+
+func Register(name string, helloFn func(func([]byte) error) error) (*Service, error) {
```

Comment:
So, as it currently stands, we have three different namespaces for messages:

1. Service Name
2. Type (on envelope)
3. URL (on UIChannel)

It seems like 1 should be enough.  What's the reason for having all 3?